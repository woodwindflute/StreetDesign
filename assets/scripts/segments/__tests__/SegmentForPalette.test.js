/* eslint-env jest */
import React from 'react'
import { render } from '../../../../test/helpers/render'
import SegmentForPalette from '../SegmentForPalette'
import { getVariantInfoDimensions } from '../view'

jest.mock('../view')

describe('SegmentForPalette', () => {
  it('renders width correctly depending on the dimension', () => {
    const dimensions = { left: 100, right: 200 }
    getVariantInfoDimensions.mockImplementation(() => dimensions)

    const { asFragment } = render(
      <SegmentForPalette type="" variantString="" randSeed={42} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
