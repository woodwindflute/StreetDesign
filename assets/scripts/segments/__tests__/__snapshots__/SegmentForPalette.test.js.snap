// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentForPalette renders width correctly depending on the dimension 1`] = `
<DocumentFragment>
  <li
    class="segment segment-in-palette"
    style="width: 420px;"
  >
    <div
      style="height: 80px;"
      tabindex="0"
    >
      <canvas
        class="segment-image"
        height="445"
        style="width: 420px; height: 445px; left: 400px;"
        width="420"
      />
    </div>
  </li>
</DocumentFragment>
`;
